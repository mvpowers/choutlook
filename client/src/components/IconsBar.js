import React from 'react';
import styled from 'styled-components';
import {
  FaCheckSquareO,
  FaEnvelopeO,
  FaCalendar,
  FaGroup,
} from 'react-icons/lib/fa';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  background-color: #f4f4f4;
  color: #005a9e;
`;

const Icon = styled.div`
  padding: 15px;

  &:hover {
    background-color: #c7e0f4;
  }
`;

export default () => (
  <Container>
    <Icon>
      <FaEnvelopeO size={16} />
    </Icon>
    <Icon>
      <FaCalendar size={16} />
    </Icon>
    <Icon>
      <FaGroup size={16} />
    </Icon>
    <Icon>
      <FaCheckSquareO size={16} />
    </Icon>
  </Container>
);
