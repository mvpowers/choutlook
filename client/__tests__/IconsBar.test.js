import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import IconsBar from '../src/components/IconsBar';

Enzyme.configure({ adapter: new Adapter() });

describe('IconsBar', () => {
  it('should be defined', () => {
    expect(IconsBar).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = shallow(<IconsBar />);
    expect(tree).toMatchSnapshot();
  });
});
