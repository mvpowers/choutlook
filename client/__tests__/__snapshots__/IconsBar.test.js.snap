// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconsBar should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <IconsBar__Icon>
          <FaEnvelopeO
            size={16}
          />
        </IconsBar__Icon>,
        <IconsBar__Icon>
          <FaCalendar
            size={16}
          />
        </IconsBar__Icon>,
        <IconsBar__Icon>
          <FaGroup
            size={16}
          />
        </IconsBar__Icon>,
        <IconsBar__Icon>
          <FaCheckSquareO
            size={16}
          />
        </IconsBar__Icon>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaEnvelopeO
            size={16}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaCalendar
            size={16}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaGroup
            size={16}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <FaCheckSquareO
            size={16}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "size": 16,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <IconsBar__Icon>
            <FaEnvelopeO
              size={16}
            />
          </IconsBar__Icon>,
          <IconsBar__Icon>
            <FaCalendar
              size={16}
            />
          </IconsBar__Icon>,
          <IconsBar__Icon>
            <FaGroup
              size={16}
            />
          </IconsBar__Icon>,
          <IconsBar__Icon>
            <FaCheckSquareO
              size={16}
            />
          </IconsBar__Icon>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaEnvelopeO
              size={16}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaCalendar
              size={16}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaGroup
              size={16}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <FaCheckSquareO
              size={16}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "size": 16,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
